Documentation/gettingstarted/tls-visibility.rst

```diff
@@ -0,0 +1,308 @@
+
+TODO:  install instructions for openssl cmd-line utility?
+
+1) Create internal Certificate Authority.
+
+Generate CA private key named 'myCA.key':
+
+.. parsed-literal::
+
+    $ openssl genrsa -des3 -out myCA.key 2048
+
+Enter any password, just remember it for some of the later steps.
+
+Generate CA certificate from the private key:
+
+.. parsed-literal::
+
+    $ openssl req -x509 -new -nodes -key myCA.key -sha256 -days 1825 -out myCA.crt
+
+The values you enter for each prompt do not need to be any specific value, and do not need to be
+accurate.
+
+2) Generate an internal private key and certificate signing with a common name that matches the DNS name
+of the destination service to be intercepted for inspection.
+
+First create the private key:
+
+.. parsed-literal::
+
+    $ openssl genrsa -out internal-swapi.key 2048
+
+Next, create a certificate signing request, specifying the DNS name of the destination service
+for the common name field when prompted.  All other prompts can be filled with any value.
+
+.. parsed-literal::
+
+    $ openssl req -new -key internal-swapi.key -out internal-swapi.csr
+
+The only field that must be a specific value is ensuring that ``Common Name'' is the exact DNS
+destination ``swapi.co'' that will be provided to the client.
+
+
+This example workflow will work for any DNS
+name as long as the toFQDN rule in the policy YAML (below) is also updated to match the DNS name in the certificate.
+
+
+3) Use the internal CA private key to create a signed certificate for swapi.co named 'internal-swapi.crt .
+
+.. parsed-literal::
+
+    $ openssl x509 -req -days 360 -in internal-swapi.csr -CA myCA.crt -CAkey myCA.key -CAcreateserial -out internal-swapi.crt -sha256
+
+Next we create a Kubernetes secret that includes both the private key and signed certificates for the destination service:
+
+.. parsed-literal::
+
+    $ kubectl create secret tls swapi-tls-data -n kube-system --cert=internal-swapi.crt --key=internal-swapi.key
+
+4) Add the CA certificate as a trusted CA inside the client pod
+
+Once the CA certificate is inside the client pod, we still must make sure that the CA file is picked up by the TLS library used by your
+application.  Most Linux applications automatically use a set of trusted CA certificates that are bundled along with the Linux distro.
+In this guide, we are using an Ubuntu container as the client, and so will update it with Ubuntu specific instructions.  Other Linux distros
+will have different mechanisms.  Also, individual applications may leverage their own certificate stores rather than use the OS certificate
+store.  Java applications and the aws-cli are two common examples.  Please refer to the application or application runtime documentation
+for more details.
+
+For Ubuntu, we first copy the additional CA certificate to the client pod filesystem
+
+.. parsed-literal::
+
+    $ kubectl cp myCA.crt default/mediabot:/usr/local/share/ca-certificates/myCA.crt
```

Comment:
One other option is to update the `myCA.crt` as a configmap and load it as a volume for all pods that will connect to such CA.

Reply:
In this case, the exact location that a cert needs to be added, and the commands that need to be run afterward vary according to different container OSes, and even in some cases for different languages.   Figuring out a better strategy around automatic CA injection would be great, but is probably beyond the scope of this getting started doc.